gandi_tf/main.py

```diff
@@ -87,33 +148,71 @@ def generate_tf(domain, entries):
+        with click.open_file(filename_tfimport, "w") as fi:
+            for cmd in commands:
+                fi.write(f"{cmd}\n")
     return commands
 
 
 @click.command(no_args_is_help=True)
 @click.option("--version", help="Display version information", is_flag=True)
+@click.option("--organization_id", help="Gandi organization ID (passing an organization ID will limit query on domain own by organization)", default=None)
+@click.option("--list_domains", help="Fetch and display fqdn_unicode", is_flag=True)
+@click.option("--list_domains_details", help="Fetch and display all domain details", is_flag=True)
+@click.option("--list_organizations", help="Fetch and display all organizations details", is_flag=True)
+@click.option('--nsfilters', help="Filter domaine based on their current nameserver 'abc','livedns' or 'other'. You can add multiple filter Example: gandi2tf --list_domains --nsfilters abc --nsfilters livedns)", multiple=True, default=[
+'abc','livedns','other'])
+@click.option("--auto_generate", help="Auto-generate tf resource based on domain from gandi instead of STDIN/USERINPUT (option can be used with --organization_id, --subdir and multiple --nsfilters)", is_flag=True)
+@click.option("--subdir", help="Create a sub-directory to store generated domain tf and tf import command", is_flag=True)
 @click.argument("domains", nargs=-1)
-def generate(domains, version):
+def generate(domains, version, organization_id, list_domains, list_domains_details, list_organizations, nsfilters, auto_generate, subdir):
     """
     Command to read Gandi.net live DNS records and generate
     corresponding TF gandi_livedns_record resources.
+
+    Warning: nameserver type 'other' and 'abc' can't be managed via terraform (for abc domain you can transfer them to livedns on gandi webinterface)
     """
+    for filter in nsfilters:
+        if filter not in ['abc','livedns','other']:
+            print(f"nsfilters error available filter: 'abc', 'livedns' or 'other' you submitted {nsfilters}")
+            return
     if version:
         import importlib.metadata
 
         _version = importlib.metadata.version("gandi-2-terraform")
         click.echo(f"Version {_version}")
         return
-
+    if list_domains:
+        print_domains_list(fetch_domains_list(organization_id), list_domains_details, nsfilters)
+        return
+    if list_domains_details:
+        print_domains_list(fetch_domains_list(organization_id), list_domains_details, nsfilters)
+        return
+    if list_organizations:
+        print_organizations_list(fetch_organizations_list())
+        return
+    if auto_generate:
+        domains = []
+        fetch_domains = fetch_domains_list(organization_id)
+        for domain in fetch_domains:
+            if domain['nameserver']['current'] in nsfilters:
+                if "other" in domain['nameserver']['current']:
+                    print("Your are trying to auto_generate a tf from a domain not managed by Gandi check tour nsfilters")
```

Comment:
There's a typo `...by Gandi check tour nsfilters`. Use also [click.echo](https://click.palletsprojects.com/en/8.1.x/utils/#printing-to-stdout) or [click.secho](https://click.palletsprojects.com/en/8.1.x/utils/#ansi-colors) to output messages. This is a general comment for all the print statements that were added.

Reply:
Hi @marcaurele,

Done